### DISCUSSION TITLE
blog/angular-inject-and-injection-functions

### DISCUSSION THREAD

**Discussion by yjaaidi** [marmicode/marmicode#112] [General]
Let's talk about https://marmicode.io/blog/angular-template-code-coverage-and-future-proof-testing

**Comment by kimpixel** [comment]
Thanks for your article.
Coverage for templates is kind of a “must-have” feature for me.
I'm surprised that this topic gets so little attention.

After making a few adjustments to my setup, I got all the tests running with AOT.

What I find really unfortunate is that it doesn't work with the Monocart Coverage Reporter.
I've grown very fond of it!

Is there anything that can be done about this?

**Comment by fagia** [comment]
Thanks for this post, it's really insightful!
As you mentioned, shallow testing is more challenging, indeed.

I'm trying to adopt the workaround you've described to avoid `TestBed#overrideComponent` and use `vi.mock` instead...
I'm not sure what I'm doing wrong here, but the component's dependencies are not getting replaced with the mocks.

My current guess is that the path I'm passing as the first argument of `vi.mock` is not correct.

In your example, it seems to be a relative path, but in my case it doesn't work: the "real" dependencies and not the mocked ones are getting picked by the test.
Could it be because my project is based on Angular Workspace (ie: multiple modules under the "projects" root directory)?
The component under test and its dependencies are anyway from the same module.

**Reply by fagia** [reply]
Ouch just found your enhancement request here: https://github.com/angular/angular-cli/issues/31609
...I guess this means that, as of now, the vi.mock approach cannot work, is it?
